yt_dlp/update.py

```diff
@@ -337,20 +468,15 @@ def update(self):
             return self._report_error('Unable to remove the old version')
 
         try:
-            newcontent = self._download(self.release_name, self._tag)
+            newcontent = self._download_asset(update_info.binary_name, update_info.tag)
         except network_exceptions as e:
             if isinstance(e, HTTPError) and e.status == 404:
                 return self._report_error(
-                    f'The requested tag {self._label(self.target_channel, self.target_tag)} does not exist', True)
-            return self._report_network_error(f'fetch updates: {e}')
+                    f'The requested tag {self.requested_repo}@{update_info.tag} does not exist', True)
+            return self._report_network_error(f'fetch updates: {e}', tag=update_info.tag)
 
-        try:
-            expected_hash = self.release_hash
-        except Exception:
-            self.ydl.report_warning('no hash information found for the release')
-        else:
-            if hashlib.sha256(newcontent).hexdigest() != expected_hash:
-                return self._report_network_error('verify the new executable')
+        if hashlib.sha256(newcontent).hexdigest() != update_info.checksum:
+            return self._report_network_error('verify the new executable', tag=update_info.tag)
```

Comment:
move `if not update_info.checksum` lines from above to down here:
```suggestion
        if not update_info.checksum:
            self._block_restart('Automatically restarting into unverified builds is disabled for security reasons')
        elif hashlib.sha256(newcontent).hexdigest() != update_info.checksum:
            return self._report_network_error('verify the new executable', tag=update_info.tag)
```